File: services/typescript/hono/src/routes/handlers/invalid_method.ts
import type { Handler } from "hono";
import { AppErrors, sendError } from "../errors.js";
import { getLogger } from "telemetry";

export const invalidMethodHandler: Handler = (c) => {
  getLogger().warn(
    { method: c.req.method, path: c.req.path },
    "Invalid method"
  );
  return sendError(c, AppErrors.InvalidMethod);
};

File: services/python/fastapi/src/routes/handlers/invalid_method.py
import structlog
from fastapi import Request
from fastapi.responses import JSONResponse

from routes.errors import AppErrors, send_error


async def invalid_method_handler(request: Request, exc: Exception) -> JSONResponse:
    structlog.get_logger().warn(
        "Invalid method", method=request.method, path=request.url.path
    )

    return send_error(AppErrors.InvalidMethod)
